internal/promapi/failover.go

```diff
@@ -176,10 +176,10 @@ func (fg *FailoverGroup) Metadata(ctx context.Context, metric string) (metadata
 	for _, prom := range fg.servers {
 		uri = prom.safeURI
 		metadata, err = prom.Metadata(ctx, metric)
-		if err == nil {
+		if err == nil && metadata != nil {
 			return metadata, nil
 		}
-		if !IsUnavailableError(err) {
+		if metadata != nil {
```

Comment:
When is there an error and metadata?

Reply:
When the first prom fails, but the second prom returns metadata, we still have error from prom 1. We are letting function caller be aware of any errors encountered, even though the promql/counter check doesn't use error directly.